internal/handler/common/update.go

```diff
@@ -1361,10 +1276,8 @@ func validateCurrentDateExpression(command string, update *types.Document) error
 					)
 				}
 			}
-			currentDateType, err := setValue.Get("$type")
-			if err != nil { // ok, default is date
-				continue
-			}
+
+			currentDateType, _ := setValue.Get("$type")
```

Comment:
Why was this changed? What happens if `$type` isn't set?

Reply:
The previous implementation is incorrect. When `setValue` is a document, then `$type` field is mandatory. If not set, we should throw an error (as done in L1284). Added new test case - `TestUpdateFieldCompatCurrentDate/NoType` to verify this.